pulpcore/cli/core/generic.py

```diff
@@ -22,27 +23,31 @@
 # Generic reusable commands
 
 
-def _task_group_filter_callback(
-    ctx: click.Context, param: click.Parameter, value: Optional[str]
-) -> Optional[str]:
-    if value is not None:
-        pulp_ctx = ctx.find_object(PulpContext)
-        assert pulp_ctx is not None
-
-        # Example: "/pulp/api/v3/task-groups/a69230d2-506e-44c7-9c46-e64a905f85e7/"
-        match = re.match(
-            rf"({pulp_ctx.api_path}task-groups/)?"
-            r"([0-9a-f]{8})-?([0-9a-f]{4})-?([0-9a-f]{4})-?([0-9a-f]{4})-?([0-9a-f]{12})/?",
-            value,
-        )
-        if match:
-            value = "{}task-groups/{}-{}-{}-{}-{}/".format(
-                pulp_ctx.api_path, *match.group(2, 3, 4, 5, 6)
+class HrefOrUuidCallback:
+    def __init__(self, base_href: str) -> None:
+        self.base_href = base_href
+
+    def __call__(
+        self, ctx: click.Context, param: click.Parameter, value: Optional[str]
+    ) -> Optional[str]:
+        if value is not None:
+            pulp_ctx = ctx.find_object(PulpContext)
+            assert pulp_ctx is not None
+
+            # Example: "/pulp/api/v3/tasks/a69230d2-506e-44c7-9c46-e64a905f85e7/"
+            href_match = re.match(
+                rf"({pulp_ctx.api_path}{self.base_href}/)?"
+                r"([0-9a-f]{8})-?([0-9a-f]{4})-?([0-9a-f]{4})-?([0-9a-f]{4})-?([0-9a-f]{12})/?",
+                value,
             )
-        else:
-            raise click.ClickException(_("Either an href or a UUID must be provided."))
+            if href_match:
+                value = "{}{}/{}-{}-{}-{}-{}/".format(
+                    pulp_ctx.api_path, self.base_href, *href_match.group(2, 3, 4, 5, 6)
```

Comment:
Is matching groups like this even necessary? Would it make sense to have one group `[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}` which will be projected to the formatted string?

Reply:
I think the regex accepts the uuid with and without the dashes. That's why we need to do it that way, no?

Reply:
Ah, I overlooked those question marks. Yes. You are right.